include/envoy/secret/secret_manager.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <fmt/format.h>
+
+#include <memory>
+#include <string>
+
+#include "envoy/secret/secret.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * A manager for static secrets.
+ *
+ * TODO(jaebong) Support dynamic secrets.
+ */
+class SecretManager {
+public:
+  virtual ~SecretManager() {}
+
+  /**
+   * add or update secret grouped by type.
+   * @param secret a shared_ptr of an implementation of Secret.
+   */
+  template <typename T> void addOrUpdateSecret(const SecretSharedPtr& secret) {
```

Comment:
-1 to this heavy use of RTTI:
https://google.github.io/styleguide/cppguide.html#Run-Time_Type_Information__RTTI_

You should be able to add a type() method returning an string (or better enum) to Secret interface and just use it from here.

Reply:
Created enum SecretType and removed RTTI